### PR TITLE
add a faq document for TIC

### PR DESCRIPTION
**Pull request by jay4peace**
### Change Description ###

add a Frequently Asked Questions for TIC

### CODE REVIEW

File: docs/platform/reference/tic-faq.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  So, you can make same binaries as the Tizen public binaries.
+
+- Regarding to the advanced mode, you can combine packages as you want by selecting building blocks. At first, you can select the architecture in the `Setting` tab,
+  And then, you can select building block packages in order by category and step by step in `Packages` tab.
+
+## What is the building block ?
+
+- The building block is the package which defines a set of combined packages for specific propose.
+  You can see pre-defined building blocks and the structure of them in [the repository of building blocks](https://git.tizen.org/cgit/tools/building-blocks/tree/?h=tizen).
+  You also can define by yourself your own blocks by creating `.inc` files and add them to [building-block.spec](https://git.tizen.org/cgit/tools/building-blocks/tree/packaging/building-blocks.spec?h=tizen) file
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:23]:
```suggestion
  You can also define your own blocks by creating `.inc` files and adding them to the [building-block.spec](https://git.tizen.org/cgit/tools/building-blocks/tree/packaging/building-blocks.spec?h=tizen) file.
```

File: docs/platform/reference/tic-faq.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## How to get a recipe for TIC ?
+
+- You can get the reference recipe by clicking the `Export` button in the advanced mode in the `Setting` tab, or by clicking the `Recipe` button in the `Images` tab if you have binaries already built.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:27]: Sentence is a little ambiguous, by "binaries already built" do you mean "pre-built binaries"?

```suggestion
- You can get the reference recipe by clicking the `Export` button in "Advanced" mode in the `Settings` tab, or by clicking the `Recipe` button in the `Images` tab if you have binaries already built.
```

File: docs/platform/reference/tic-faq.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## How to add my applications to the Tizen binary ?
+
+- You can add your application packages(`tpk` packages) in `setting` tab in both of the easy and advanced mode by clicking the `import` button and selecting the radio button for `tpk` file.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:31]:
```suggestion
- You can add your application packages (`tpk` packages) in the `Settings` tab in both "Easy" and "Advanced" mode by clicking the `Import` button and selecting the radio button for `tpk` file.
```

File: docs/platform/reference/tic-faq.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## How to add my platform packages to the Tizen binary ?
+
+- You can your platform module packages(`rpm` packages) in `setting` tab in the advanced mode, by clicking the `import` button and selecting the radio button for `rpm` file . But, to add your `rpm` packages to the binary, you have to check them in the tree view of the `Packages` tab.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:35]:
```suggestion
- You can add your platform module packages (`rpm` packages) in the `Settings` tab in "Advanced" mode, by clicking the `Import` button and selecting the radio button for `rpm` file. However, in order to add your `rpm` packages to the binary, you have to check them in the tree view of the `Packages` tab.
```

File: docs/platform/reference/tic-faq.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## Can I find artifacts of TIC in my host PC ?
+
+- You can find them in the `tif-artifaces` directory in your home directory, if you use `docker-compose`(https://s3-us-west-1.amazonaws.com/tizenschool/257/docker-compose.yaml) file provided by us.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:39]: Should it be "tif-artifaces" or "tif-artifacts"? The heading says "artifacts" so I'm uncertain here. If the correct term is "artifaces", should the heading be changed as well?
```suggestion
- You can find them in the `tif-artifaces` directory in your home directory, if you use the `docker-compose` file  (https://s3-us-west-1.amazonaws.com/tizenschool/257/docker-compose.yaml) provided by us.
```

**Review by asm-foysal — COMMENTED** [review]
In the file docs/platform/developing/creating-tic.md 
I wanted to provide several comments:
Line 5: "Set up docker" not "Settings up Docker"
 >Unable to provide comment.

Line 17: "higher" not "Higher"
 >Unable to provide comment.

Line 33: <http://127.0.0.1:8088> -> Link does not work
 >Unable to provide comment.

File: docs/platform/reference/tic-faq.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+# TIC Frequently Asked Questions
+
+- [What is for the easy mode and the advanced mode ?](#what-is-for-the-easy-mode-and-the-advanced-mode-)
+- [What is the building block ?](#what-is-the-building-block-)
+- [How to get a recipe for TIC ?](#how-to-get-a-recipe-for-tic-)
+- [How to add my application to the Tizen binary ?](#how-to-add-my-application-to-the-tizen-binary-)
+- [How to add my platform packages to the Tizen binary ?](#how-to-add-my-platform-packages-to-the-tizen-binary-)
+- [Can I find artifacts of TIC in my host PC ?](#can-i-find-artifacts-of-tic-in-my-host-pc-)
+
+## What is for the easy mode and the advanced mode ?
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:11]: Same here, please consider revising the sentence.
```suggestion
## What are the "Easy" and "Advanced" modes ?
```

**Review by asm-foysal — COMMENTED** [review]
Made some changes to the previous comments based on Sagar's suggestions.

File: docs/platform/developing/creating-tic.md [hunk 7]
```diff
@@ -35,3 +35,5 @@ Open the web browser and enter <http://127.0.0.1:8088> in the address bar. The T
   ![TIC web-service](./media/tic-ui.png)
```
[the remark this answers is not in the document]
Comment (jay4peace) [reply docs/platform/developing/creating-tic.md:35]: fixed
http://127.0.0.1:8088 is not a link, it is a url for local host installed TIC

[the remark this answers is not in the document]
Comment (asm-foysal) [reply docs/platform/developing/creating-tic.md:35]: @jay4peace thank you for the fix. I'll mark the conversation as resolved.

File: docs/platform/reference/tic-faq.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+- The building block is the package which defines a set of combined packages for a specific propose.
+  You can see pre-defined building blocks and their structure in [the repository of building blocks](https://git.tizen.org/cgit/tools/building-blocks/tree/?h=tizen).
+  You can also define by yourself your own blocks by creating `.inc` files and adding them to the [building-block.spec](https://git.tizen.org/cgit/tools/building-blocks/tree/packaging/building-blocks.spec?h=tizen) file.
+
+## What are the "Easy" and "Advanced" mode ?
+
+- If you select "Easy" mode, you can create exactly the same binaries as the officially released [Tizen binaries](http://download.tizen.org/releases/milestone/tizen/unified/latest/images/standard/). Also, you can create your own binary based on the official one by importing your own packages in this mode.
+
+- On the other hand, in "Advanced" mode, you can customize your own binary from the scratch. At first, you can select the architecture in the `Settings` tab,
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:21]: You don't need a defining article here.
```suggestion
- On the other hand, in "Advanced" mode, you can customize your own binary from scratch. At first, you can select the architecture in the `Settings` tab,
```

File: docs/platform/reference/tic-faq.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  then, you can select the building block packages one by one in the `Packages` tab.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:22]: No comma necessary here.
```suggestion
  then you can select the building block packages one by one in the `Packages` tab. 
```

File: docs/platform/reference/tic-faq.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+- You can get the reference recipe by clicking the `Export` button in "Advanced" mode in the `Settings` tab, or by clicking the `Recipe` button in the `Images` tab if you have binaries already built.
+
+## How to add my applications to the Tizen binary ?
+
+- You can add your application packages (`tpk` packages) in the `Settings` tab in both "Easy" and "Advanced" mode by clicking the `Import` button and selecting the radio button for `tpk` file.
+
+## How to add my platform packages to the Tizen binary ?
+
+- You can add your platform module packages (`rpm` packages) in the `Settings` tab in "Advanced" mode, by clicking the `Import` button and selecting the radio button for `rpm` file . However, in order to add your `rpm` packages to the binary, you have to check them in the tree view of the `Packages` tab.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:36]:
```suggestion
- You can add your platform module packages (`rpm` packages) in the `Settings` tab in "Advanced" mode. Just click the `Import` button and select the radio button for `rpm` file. However, in order to add your `rpm` packages to the binary, you have to check them in the tree view of the `Packages` tab.
```

File: docs/platform/reference/tic-faq.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+## Can I find artifacts of TIC in my host PC ?
+
+- You can find them in the `tic-artifacts` directory in your home directory, if you use the [`docker-compose`](https://s3-us-west-1.amazonaws.com/tizenschool/257/docker-compose.yaml) file provided by us.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:40]: The text "docker-compose" should show up as a link only, it's difficult to notice when the text is highlighted as well. 
```suggestion
- You can find them in the `tic-artifacts` directory in your home directory, if you use the [docker-compose](https://s3-us-west-1.amazonaws.com/tizenschool/257/docker-compose.yaml) file provided by us.
```

**Review by asm-foysal — COMMENTED** [review]
@jay4peace  The navigation links at the very top of the page is not working. The ones on the right are ok. Please address the top links.

**Review by asm-foysal — COMMENTED** [review]
@jay4peace I'm done with my final review; I just found some grammatical and spelling errors that need to be addressed. Please take care of these and we should be good to go.

**Comment by jay4peace** [comment]
@asm-foysal Thank you for your kindness review.

File: docs/platform/reference/tic-faq.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  You can see pre-defined building blocks and their structure in [the repository of building blocks](https://git.tizen.org/cgit/tools/building-blocks/tree/?h=tizen).
+  You can also define by yourself your own blocks by creating `.inc` files and adding them to the [building-block.spec](https://git.tizen.org/cgit/tools/building-blocks/tree/packaging/building-blocks.spec?h=tizen) file.
+
+## What are the "Easy" and "Advanced" mode?
+
+- If you select "Easy" mode, you can create exactly the same binaries as the officially released [Tizen binaries](http://download.tizen.org/releases/milestone/tizen/unified/latest/images/standard/). Also, you can create your own binary based on the official one by importing your own packages in this mode.
+
+- On the other hand, in "Advanced" mode, you can customize your own binary from scratch. At first, you can select the architecture in the `Settings` tab,
+  then you can select the building block packages one by one in the `Packages` tab.
+  We recommand you select a building block in the `Preset` category first, and the `Domain` category later.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:23]: Spelling error needs to be addressed.
```suggestion
  We recommend you select a building block in the `Preset` category first, and the `Domain` category later.
```

File: docs/platform/reference/tic-faq.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+- [How to get a recipe for TIC?](#how-to-get-a-recipe-for-tic)
+- [How to add my applications to the Tizen binary?](#how-to-add-my-applications-to-the-tizen-binary)
+- [How to add my platform packages to the Tizen binary?](#how-to-add-my-platform-packages-to-the-tizen-binary)
+- [Can I find artifacts of TIC in my host PC?](#can-i-find-artifacts-of-tic-in-my-host-pc)
+
+## What is the building block?
+
+- The building block is the package that defines a set of combined packages for a specific purpose.
+  You can see pre-defined building blocks and their structure in [the repository of building blocks](https://git.tizen.org/cgit/tools/building-blocks/tree/?h=tizen).
+  You can also define by yourself your own blocks by creating `.inc` files and adding them to the [building-block.spec](https://git.tizen.org/cgit/tools/building-blocks/tree/packaging/building-blocks.spec?h=tizen) file.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:15]: Just noticed this, the "by yourself" part is redundant so suggesting you to removing it.
```suggestion
  You can also define your own blocks by creating `.inc` files and adding them to the [building-block.spec](https://git.tizen.org/cgit/tools/building-blocks/tree/packaging/building-blocks.spec?h=tizen) file.
```

File: docs/platform/reference/tic-faq.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+- You can add your application packages (`tpk` packages) in the `Settings` tab in both "Easy" and "Advanced" mode by clicking the `Import` button and selecting the radio button for `tpk` file.
+
+## How to add my platform packages to the Tizen binary?
+
+- You can add your platform module packages (`rpm` packages) in the `Settings` tab in "Advanced" mode. Just click the `Import` button and select the radio button for `rpm` file. However, in order to add your `rpm` packages to the binary, you have to check them in the tree view of the `Packages` tab.
+
+## Can I find artifacts of TIC in my host PC?
+
+- You can find them in the `tic-artifacts` directory in your home directory, if you use the [docker-compose](https://s3-us-west-1.amazonaws.com/tizenschool/257/docker-compose.yaml) file provided by us.
```
Comment (asm-foysal) [inline docs/platform/reference/tic-faq.md:40]: The comma is not necessary here.
```suggestion
- You can find them in the `tic-artifacts` directory in your home directory if you use the [docker-compose](https://s3-us-west-1.amazonaws.com/tizenschool/257/docker-compose.yaml) file provided by us.
```

**Review by asm-foysal — COMMENTED** [review]
@jay4peace I noticed some minor errors in the text, so I've suggested corrections for them. Please address these. Everything else looks great!

**Review by asm-foysal — APPROVED** [review]
@jay4peace thanks for making all the requested changes, Please fix the reference link "platform/reference/set-up-docker/" since it's now showing up in the ToC.

File: docs/platform/developing/creating-tic.md [hunk 15]
```diff
@@ -2,7 +2,7 @@
 
 This topic provides information on how to create a Tizen image with TIC.
 
-To use TIC, you have to set up the docker initially. For information on docker setup, see [Setting up Docker](../reference/setting-up-docker.md).
+To use TIC, you have to set up the docker initially. For information on docker setup, see [Set up docker](../reference/set-up-docker.md).
```
Comment (asm-foysal) [inline docs/platform/developing/creating-tic.md:5]: The reference link is not showing up in the ToC.

**Review by asm-foysal — CHANGES_REQUESTED** [review]

File: docs/platform/developing/creating-tic.md (hunk 15 above)
Comment (annie-samsung) [reply docs/platform/developing/creating-tic.md:5]: This link is working so I am resolving this conversation.

File: docs/platform/reference/tic-faq.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
+<!-- omit in toc -->
+# TIC Frequently Asked Questions
```
Comment (annie-samsung) [inline docs/platform/reference/tic-faq.md:2]: I see that the FAQ page format is different from [https://docs1.stg.tizen.org/staging/1564/platform/reference/gbs/gbs-faq/](https://docs1.stg.tizen.org/staging/1564/platform/reference/gbs/gbs-faq/) and MIC Faq, however I like this page's format so I guess we can keep it as is.

File: docs/platform/toc_all.md [hunk 17]
```diff
@@ -75,7 +75,8 @@
 ### [Modifying Certificate Manager Texts](/platform/reference/tizen-studio/example-web-certi-ext.md)
 ### [Modifying Eclipse Themes](/platform/reference/tizen-studio/example-web-theme-ext.md)
 
-## [Setting up docker](/platform/reference/setting-up-docker.md)
+## [Tizen Image Creator FAQ](/platform/reference/tic-faq.md)
+## [Set up docker](/platform/reference/set-up-docker.md)
```
Comment (annie-samsung) [inline docs/platform/toc_all.md:79]: @asm-foysal, whenever we change the file name we check if the old file name is referenced in any other page and change them as well. @sagarkedari could you help him on how to check on the same.

**Review by annie-samsung — APPROVED** [review]
Please find my comments

File: docs/platform/toc_all.md (hunk 17 above)
Comment (asm-foysal) [reply docs/platform/toc_all.md:79]: image
@annie-samsung, I discussed this with Sagar and got the hang of checking if a removed/renamed file was referenced in any other pages. It appears the "setting-up-docker" was referenced only on these files so we should be fine.

**Comment by annie-samsung** [comment]
@asm-foysal could you approve this PR, so that I can merge it.

**Review by asm-foysal — APPROVED** [review]
@annie-samsung , the final review is done and the PR is good to go, I'm approving this PR for merging.

**Pull request merged**